text/0001-cdk-update.md

```diff
@@ -0,0 +1,293 @@
+---
+rfc pr: [#335](https://github.com/aws/aws-cdk-rfcs/pull/335)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/1
+---
+
+# CDK Update
+
+Rapid and automatic updates for code-only changes to CDK application stacks.
+
+## Working Backwards
+
+### CHANGELOG
+
+- feat(cli): implement `cdk update` and `cdk update --watch` for asset updates
+
+### README
+
+The `cdk update` command accelerates the asset deployments for your CDK
+application during development by inspecting the assets in your application,
+identifying those that can be updated in-place without a full CDK deployment,
+and doing so. It can either do this as an interactive command or by running
+continuously and monitoring the input files.
+
+For supported construct types (see list, below) the update command will identify
+assets whose content has changed since the publication of the CDK, assemble them
+from sources if necessary, and then publish them to their asset storage medium.
+It will then use AWS SDK commands to directly modify the CDK resources to use
+the new asset paths. The update command acts on one or more stacks in the
+application, and will only perform the shortcut if all changes are updateable.
+
+A stack is "updateable" if the stack is explicitly marked as updateable, and it
+contains any assets supported by the update process. At this time, only Lambda
+and ECS images are supported, and so any stack with one of those is updateable.
+
+At a high level, `update` operates by reading the current state of the stack,
+performing a stack diff on it, and if the stack is different only in resources
+supported by update plugins, running those in lieu of a full deployment.
+
+#### Preparing Your Stack
+
+In order to ensure that update is not invoked against production stacks, you
+must explicitly designate your stack as being updateable in your application.
+This is a stack-level property, set in the `StackProps` parameter to the stack
+constructor:
+
+```typescript
+const myDevelopmentStack = new ApplicationStack(app, "StackNameForDev", {
```

Comment:
Since I don't think we are going to offer a dedicated command for this, but rather support this only via watch, this is not required.